test/other_algs_test.jl

```diff
@@ -0,0 +1,48 @@
+using NeuralPDE
+
+using ModelingToolkit, Optimization, OptimizationOptimisers
+import ModelingToolkit: Interval, infimum, supremum
+import Lux: tanh, identity
+
+@testset begin
```

Comment:
Can you add the tests that were present in the paper? - `Hamilton-Jacobi-Bellman` and `Burgers`

Reply:
The architecture they used for `Burgers` is pretty heavy, 6 LSTM type layers with 200 units per layer. Tuning and then testing would be testing the compute resources of the system too.

For `Hamilton-Jacobi-Belman`, if I remember right from the paper, `NeuralPDE.jl` currently does not support Stochastic DEs. Here, #531 is the relevant issue.

Reply:
What is the number of parameters in the Lux Model with 6 layers? If it is too large, maybe try reducing the domain size and size of the model?

Reply:
It is 995,201 parameters. On this, the paper does not provide a clean initial condition. It's a linear combination to match the two BCs. They attempt to solve for the PDE such that the coefficients are also provided. I can maybe do for a specific case if that suffices, maybe the one they do in the Raissi's PINN paper? Also, to check the tolerance in that case, is it fine if I do a finite difference solution? Sorry, I missed this in the clutter too.

Reply:
Hmm, sure. I think we should have a couple more test cases and not just Poisson's equation

Reply:
I added the European Call Option from "Solving Nonlinear and High-Dimensional Partial Differential Equations via Deep Learning" by A. Al-Aradi, A. Correia, D. Naiff, G. Jardim and Y. Saporito ([https://arxiv.org/abs/1811.08782](https://arxiv.org/abs/1811.08782)) as another test case and looks like it passes for the commit [82d7830](https://github.com/SciML/NeuralPDE.jl/pull/802/commits/82d78305d27bae6ac681521534dda0cccfa2be49). Getting a finite difference solution to Burger's equation, from Raissi's paper, had issues because the numerical solution had diffusion errors, blurring the shock interface. Also, I guess there are no other FD checks inside the `test` because of their own inaccuracies, so I'm unsure if we should add it.

Reply:
Nice! For getting solution to Burger's equation, did you try using MethodOfLines.jl - https://github.com/SciML/MethodOfLines.jl?

Reply:
`MethodOfLines` isn't listed in `Project.toml`, even as a compat package and adding it modifies the `Project.toml` file so didn't try that.

Reply:
You can add it as a test dependency.

Reply:
Done!